Description: Let ESC close the help
 It was suggested by users that it is desired for ESC to close the help
 window.
---
Origin: upstream, https://github.com/Tomas-M/iotop/commit/8ea414c53ca3b5986191d06a8f99f96282975e02 https://github.com/Tomas-M/iotop/commit/0d48498cc593d3b72d56fb5a5519b2b8612fbc31
Forwarded: not-needed
Last-Update: 2021-09-06

--- a/src/view_curses.c
+++ b/src/view_curses.c
@@ -1092,6 +1092,9 @@ static inline int curses_key(int ch) {
 			config.f.deadx=!config.f.deadx;
 			break;
 		case 27: // ESC
+			if (showhelp&&!in_ionice&&!in_filter)
+				showhelp=0;
+			// unlike help window these cannot happen at the same time
 			if (in_ionice)
 				in_ionice=0;
 			if (in_filter)
--- a/iotop.8
+++ b/iotop.8
@@ -199,6 +199,9 @@ Sort by previous column
 \fB<esc>\fR
 Cancel ionice or filter selection. In case only the help window is open then close it
 .TP
+\fB<esc>\fR
+Cancel ionice or filter selection. In case only the help window is open then close it
+.TP
 \fBo\fR, \fBO\fR
 Toggle showing only processes with IO activity
 .TP
